### ISSUE TITLE
[BUG] Topology prior probability incorrect

### ISSUE THREAD

**Issue by plewis** [revbayes/revbayes#212]
**Describe the bug**
Topology prior probability incorrect for unrooted trees (e.g. 1/15 instead of 1/105 for unrooted tree with 6 taxa)

**To Reproduce**
```
#########################################
# Set the pseudorandom number seed      #
#########################################

seed(35791)

#############################################
# Define the prior on the tree topology     #
# For 6 taxon unrooted tree, should be      #
# equal to 1/105 (-4.65396035 on log scale) #
#############################################

taxa <- v(taxon("A"),taxon("B"),taxon("C"),taxon("D"),taxon("E"),taxon("F"))
topology ~ dnUniformTopology(taxa)

################################################
# Define the prior on the branch lengths       #
# Joint prior on 9 branch lengths each         #
# of length 0.1 and having an Exponential(10)  #
# prior should be 9*log(10) - 10*.9 = 11.72327 # 
# on log scale.                                #
################################################

for (i in 1:9) {
   edgelens[i] ~ dnExponential(10.0)
   edgelens[i].setValue(0.1)
}
tree_length := sum(edgelens)

################################################
# Combine the tree topology and branch lengths #
################################################

phylogeny := treeAssembly(topology, edgelens)

############################################
# Define the substitution model parameters #
############################################

# JC model
Qmatrix := fnJC(4)

#################################
# Define the phyloCTMC model    #
# (AKA the likelihood function) #
#################################

seq  ~ dnPhyloCTMC(tree=phylogeny, Q=Qmatrix, type="DNA")

#########################
# Make the model object #
#########################

my_model = model(phylogeny)

#########################
# Run the MCMC analysis #
#########################

monitors = VectorMonitors()
monitors.append( mnScreen(printgen=1, tree_length) )

moves = VectorMoves()
analysis = mcmc(my_model, monitors, moves)
analysis.run(generations=1, underPrior=TRUE) 

#######################################################################
# Joint log prior = 7.06930965 = -4.65396035 + 11.72327               #
# However, RevBayes says 9.015219799 = -2.708050201 + 11.72327        #
#                                                                     #
# Iter        |      Posterior   |     Likelihood   |          Prior  #
# --------------------------------------------------------------------#
# 0           |        9.01522   |              0   |        9.01522  #
# 1           |        9.01522   |              0   |        9.01522  #
#######################################################################

########
# Quit #
########
q()
```
**Expected behavior**
Expected log topology prior to equal -log(105) = -4.65396035 but instead it was -log(15) = -2.708050201. Combined with the log joint branch length prior (11.72327), the total prior should be 7.06930965 but instead RevBayes says it is 9.015219799.

**Screenshots**
```
Iter        |      Posterior   |     Likelihood   |          Prior   |    tree_length   |    elapsed   |        ETA   |
-----------------------------------------------------------------------------------------------------------------------
0           |        9.01522   |              0   |        9.01522   |            0.9   |   00:00:00   |   --:--:--   |
1           |        9.01522   |              0   |        9.01522   |            0.9   |   00:00:00   |   --:--:--   |
```

**Computer info**
Which OS are you using? **MacOS 10.15.7**
Which version of RevBayes? **1.1.1**

**Additional context**
Line 35 in src/core/distributions/phylogenetics/tree/UniformTopologyDistribution.cpp is

	for (size_t i = 2; i < 2*num_taxa - 3 - 2*(!rooted); i++)

but it should be 

	for (size_t i = 2; i <= 2*num_taxa - 3 - 2*(!rooted); i++)

(i.e. the < should be an <=).

**Comment by bredelings** [comment]
Does that fix this?

I'm having trouble following the code here.

I'm used to seeing double-factorials (3*5*7...) etc. for the number of unrooted topologies, but there are only single factorials here...

It looks like this "uniform" distribution is uniform on ORDERED rooted topologies?  That is, in a tree like `((A,B)X,(C,D)Y);`, the time-order of the internal nodes X and Y is part of the rooted topology?

**Comment by plewis** [comment]
### Double factorial
The code does compute a double-factorial. The double factorial (2n-5)!! can be written in terms of single factorials: (2n-5)!/[2^(n-3) (n-3)!]. The line that needs modification is involved in computing the log of (2n-5)!

### Uniform distribution
The documentation says that the rooted argument in dnUniformTopology is FALSE by default and that is confirmed by this line in _src/core/distributions/phylogenetics/tree/UniformTopologyDistribution.h_:

`UniformTopologyDistribution(const std::vector<Taxon> &ta, const Clade &og, const std::vector<Clade> &c, bool rooted = false);`

So, in this example, the uniform distribution is over UNrooted topologies.

**Comment by davidcerny** [comment]
Fixed by PR #442.
